R/check_target_tbl_values.R

```diff
@@ -16,11 +29,44 @@ check_target_tbl_values <- function(
     "oracle-output"
   ),
   file_path,
-  hub_path
+  hub_path,
+  date_col = NULL,
+  allow_extra_dates = FALSE,
+  config_target_data = NULL
```

Comment:
I am confused about why this argument is needed if the simplest expected behavior is just that it would be loaded directly from the `hub_path`?

Reply:
This was a design pattern choice to avoid redundant file reads. Since multiple validation functions need this data, we read it once at the higher level target data validation functions (`validate_target_data()` and `validate_target_submission()`) and pass it down to the lower-level check functions that need it.

You're right that we could add fallback logic to load from `hub_path` if `config_target_data` is not provided, but we'd need to add that pattern consistently across all the check functions that use this argument. Happy to address this in a separate issue if you think the convenience outweighs the additional code, let me know!